api/src/utils/hooks.ts

```diff
@@ -52,33 +48,19 @@ export async function verifyJWT(context: HookContext) {
     return Promise.reject(new Forbidden("The authorization header is missing"));
   }
   const token = authHeader.split(" ")[1];
-
-  // fetch public key from JWKS url and verify token
-  const jwksOptions = {
-    jwksUri: context.app.get("authentication").jwksUri,
-  } as ClientOptions;
-  const client = jwks(jwksOptions) as JwksClient;
-  const keys = (await client.getSigningKeysAsync()) as SigningKey[];
-
-  let decoded;
-  try {
-    decoded = verify(token, keys[0].getPublicKey(), {
-      algorithms: context.app.get("authentication").algorithms,
-    });
-  } catch (error) {
-    throw new NotAuthenticated(`Authentication failed: ${error.message}`);
-  }
+  const keys = await getAuthSigningKeys(context);
+  decodeIdToken(token, keys, context);
   return context;
```

Comment:
Does returning the context pass the hook? Should an error be instead thrown if authentication fails?

Reply:
Yes, returning the context would pass the hook. An exception would be raised in the `verifyIdToken` function if something fails during verification and break the chain.